### DISCUSSION TITLE
Transition to RESTful web service for benchmark results (PrimeView)

### DISCUSSION THREAD

**Discussion by rbergen** [PlummersSoftwareLLC/Primes#855] [Announcements]
Hello everyone,

It's been a while since the last announcement within the boundaries of the Primes project. We have one today however, and we believe it's a good one.

For about a year now, the [PrimeView](https://plummerssoftwarellc.github.io/PrimeView) application has been available to review the benchmark results that are generated by a number of machines. These machines (runners) execute the full benchmark at a daily frequency, and post the results to a location from which PrimeView retrieves them. If you haven't opened PrimeView yet, we won't hold it against you if you take a look now, and read the rest of this later. :)

Up to a couple of days ago, the benchmark results were posted to an S3 bucket on AWS. This was a pragmatic solution that allowed us to get the daily benchmark setup working quite quickly. It served us well, but due to the effectively file-based nature of S3 storage, it also came with some clear limitations.
This is why we're happy to announce that benchmark results are now being posted to, and retrieved from, a RESTful web service. We're now in the process of monitoring the runners to make sure they post to the web service as reliably as they did to the S3 bucket.

The immediate result of this transition is that PrimeView is quite a bit snappier than it was. Also, it is now possible to freely navigate the history of benchmark results, going back to October 2021. Furthermore, this new method of collecting benchmark results makes it possible to add entirely new ways to interact with the benchmark results in the future. We have some interesting ideas about this, which we aim to start implementing in the upcoming months. Of course, we'll let you know here when there is something new to look at, so stay tuned!

We'd like to again thank everyone for contributing to and engaging with this project. We're still truly amazed by how big this has become, and we know that's a statement we can also make on behalf of @davepl. 

Best,

Tudor and Rutger

P.S. We are still open to adding additional daily runners to the ones we currently have. Please reach out to us if you'd like one of your machines to run a daily benchmark and post its results to our new web service.

**Comment by mike-barber** [comment]
Nice work guys - it does feel a lot bit snappier indeed! And also awesome to see Tudor's M1 Max run.

**Reply by marghidanu** [reply]
Thank you @mike-barber!
